core/store/src/trie/mod.rs

```diff
@@ -710,14 +773,31 @@ impl Trie {
                 assert!(remainder.is_empty());
                 let leaf_key =
                     chunks.into_iter().map(|chunk| (chunk[0] * 16) + chunk[1]).collect::<Vec<u8>>();
-                let state_record = StateRecord::from_raw_key_value(leaf_key, bytes.to_vec());
+                let state_record =
+                    StateRecord::from_raw_key_value(leaf_key.clone(), bytes.to_vec());
+
+                fn types_match(type_str: Option<&str>, type_byte: u8) -> bool {
+                    if !type_str.is_some() {
+                        return true;
+                    }
+                    let trie_key_col = col::NON_DELAYED_RECEIPT_COLUMNS
```

Comment:
note: not sure if `col::NON_DELAYED_RECEIPT_COLUMNS` is sth I should use here just because it allows me to achieve my goal